### DISCUSSION TITLE
Error: "magick: unbalanced parenthesis `(end of cli)'..."

### DISCUSSION THREAD

**Discussion by Deziminatix** [ImageMagick/ImageMagick#3709] [Help]
I have a Python script that I run using a .bat file which first runs Anaconda Prompt and then executes the Python script. Then Python script prompts a path to the file with a certain word. Then Tesseract-OCR and ImageMagick apply this word to an image. And at the finish, a picture with a word on it imports to my phone.

But right after I enter a path to the file with a certain word it shows an error:  "magick: unbalanced parenthesis `(end of cli)' at CLI arg 15 @ error/magick-cli.c/ProcessCommandOptions/505."

It's not my script, a friend of mine shared it with me and it worked great for a while, but then something happened and the error started popping up, but then I somehow fixed it without changing the Python script code and it worked great . But now I just reinstalled Windows and the error started popping up again. I'm using the latest ImageMagick version "ImageMagick-7.0.11-13-Q16-x64-static"

CMD screenshot:
![image](https://user-images.githubusercontent.com/13819963/118909191-4f4b4f80-b92b-11eb-961d-354c2b3fc232.png)

The Python script code:

```
import requests
import os
import subprocess
from PIL import Image
from pytesseract import image_to_string
import pytranslator
from string import digits

file = input("Picture file: ")

xuj = ( image_to_string(Image.open(('{}').format(file)),lang='eng')  )
xuj1 = (xuj)
pizda = (xuj)
pizda = pizda.replace(" ", "_")
pizda = 'Picture' + ".jpg"
numbers = sum(c.isdigit() for c in xuj)
numbersInt = int(numbers)
if (numbersInt) >= 5:
	remove_digits = str.maketrans('', '', digits)
	xuj = xuj.translate(remove_digits)
	print("Too many digits")
else:
	print("Less than 4 digits")

if len(xuj) <= 9:
	cmd = 'magick MilaAzul_Proof.jpg -pointsize 53 -font GoodDogPlain.ttf -background none -fill rgba(0,0,0,0.85) -size 200x ( -gravity center caption:"{}" -rotate -16 -blur 25 ) -gravity Center -geometry +110-10 -compose over -composite {}'.format(xuj, pizda)
	subprocess.call(cmd, shell=True)
	print('less then 9 chars')
	os.remove(file)

else:
	cmd = 'magick MilaAzul_Proof.jpg -pointsize 45 -font GoodDogPlain.ttf -background none -fill rgba(0,0,0,0.85) -size 255x ( -gravity center caption:"{}" -rotate -16 -blur 25 ) -gravity Center -geometry +120-10 -compose over -composite {}'.format(xuj, pizda)
	subprocess.call(cmd, shell=True)
	os.remove(file)
	print('more than 9 chars')

print("Entered word: {}".format(xuj1)) 

cmd1 = 'adb -s 01c794fbf4ab push {} /sdcard/Pictures/'.format(pizda)

subprocess.call(cmd1, shell=True)
```

**Comment by snibgo** [comment]
Please print the value of cmd after setting it.

**Reply by Deziminatix** [reply]
Hope I understood you correctly.

```C:\Users\yarak\Desktop>call C:\Users\yarak\anaconda3\Scripts\activate.bat

(base) C:\Users\yarak\Desktop>cd C:\Users\yarak\SnapchatFakeProof\Scripts\MilaAzul

(base) C:\Users\yarak\SnapchatFakeProof\Scripts\MilaAzul>script.py
Picture file: C:\Users\yarak\Documents\ShareX\Screenshots\2021-05\DimScreen_3Fh5Yefku6.png
Less than 4 digits
magick: unbalanced parenthesis `(end of cli)' at CLI arg 15 @ error/magick-cli.c/ProcessCommandOptions/505.
more than 9 chars
Entered word: someWord```

**Comment by fmw42** [comment]
Put double quotes around rgba(0,0,0,0.85). Does that help?

**Reply by Deziminatix** [reply]
Just tried. Didn't help :(

I'm almost sure the problem isn't in code but with something else... Maybe ImageMagick version is wrong...

**Comment by snibgo** [comment]
I'm not an expert at python.

I suspect the problem is that "cmd" isn't what you expect, perhaps because the parentheses need to be escaped, or something.

Please print the value of cmd. I suppose this will do it:
```
print(cmd)
```
